contracts/upgradeable_contracts/amb_erc20_to_native/BlockReward.sol

```diff
@@ -0,0 +1,178 @@
+        return addressStorage[keccak256(abi.encodePacked(EXTRA_RECEIVER_ADDRESS, _index))];
+    }
+
+    function extraReceiverAmount(address _receiver) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked(EXTRA_RECEIVER_AMOUNT, _receiver))];
+    }
+
+    function extraReceiversLength() public view returns (uint256) {
+        return uintStorage[EXTRA_RECEIVERS_LENGTH];
+    }
+
+    function mintedForAccount(address _account) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked(MINTED_FOR_ACCOUNT, _account))];
+    }
+
+    function mintedForAccountInBlock(address _account, uint256 _blockNumber) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked(MINTED_FOR_ACCOUNT_IN_BLOCK, _account, _blockNumber))];
+    }
+
+    function mintedInBlock(uint256 _blockNumber) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked(MINTED_IN_BLOCK, _blockNumber))];
+    }
+
+    function mintedTotally() public view returns (uint256) {
+        return uintStorage[MINTED_TOTALLY];
+    }
+
+    function mintedTotallyByBridge(address _bridge) public view returns (uint256) {
+        return uintStorage[keccak256(abi.encodePacked(MINTED_TOTALLY_BY_BRIDGE, _bridge))];
+    }
+
+    function _addExtraReceiver(address _receiver) private {
+        uint256 _index = extraReceiversLength();
+        addressStorage[keccak256(abi.encodePacked(EXTRA_RECEIVER_ADDRESS, _index))] = _receiver;
+        uintStorage[EXTRA_RECEIVERS_LENGTH] = _index + 1;
+    }
+
+    function _addMintedTotallyByBridge(uint256 _amount, address _bridge) private {
+        bytes32 hash = keccak256(abi.encodePacked(MINTED_TOTALLY_BY_BRIDGE, _bridge));
+        uintStorage[hash] = uintStorage[hash].add(_amount);
+    }
+
+    function _setBridgeAmount(uint256 _amount, address _bridge) private {
+        uintStorage[keccak256(abi.encodePacked(BRIDGE_AMOUNT, _bridge))] = _amount;
+    }
+
+    function _clearExtraReceivers() private {
+        uintStorage[EXTRA_RECEIVERS_LENGTH] = 0;
+    }
+
+    function _isBridgeContract(address _addr) private view returns (bool) {
+        address[bridgesAllowedLength] memory bridges = bridgesAllowed();
+
+        for (uint256 i = 0; i < bridges.length; i++) {
+            if (_addr == bridges[i]) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    function _setExtraReceiverAmount(uint256 _amount, address _receiver) private {
+        uintStorage[keccak256(abi.encodePacked(EXTRA_RECEIVER_AMOUNT, _receiver))] = _amount;
+    }
+
+    function _setMinted(uint256 _amount, address _account) private {
+        bytes32 hash = keccak256(abi.encodePacked(MINTED_FOR_ACCOUNT_IN_BLOCK, _account, block.number));
+        uintStorage[keccak256(abi.encodePacked(MINTED_FOR_ACCOUNT_IN_BLOCK, _account, block.number))] = _amount;
```

Comment:
can we use `hash` calculated above?

Reply:
Yes, of course. That was simply a copy-paste mistake.
Fixed it.